### PR TITLE
[MOB-15980] Fixed getTrackingIdentifier and getVisitorIdentifier to return nil string and not return unexpected error when aid vid values are not found.

### PR DESCRIPTION
**Pull request by addb**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by emdobrin** [comment]
I am considering an alternative solution, to modify getSharedState to include empty values when aid/vid are not available, so the shared state and response identity event shares the keys verbosely like:
```
Standard shared state created for com.adobe.module.analytics with version x and data: 
{
  "aid" : "",
  "vid" : ""
}
```
This option is also in line with Android and ACP versions. @kevinlind @addb thoughts?

**Comment by kevinlind** [comment]
> I am considering an alternative solution, to modify getSharedState to include empty values when aid/vid are not available, so the shared state and response identity event shares the keys verbosely like:
> 
> ```
> Standard shared state created for com.adobe.module.analytics with version x and data: 
> {
>   "aid" : "",
>   "vid" : ""
> }
> ```
> 
> This option is also in line with Android and ACP versions. @kevinlind @addb thoughts?

Honestly, I prefer to filter out null/empty values from the EventData. I think the Swift implementation actually corrects the C++ implementation, so I'd like to leave the shared state as it is.

**Comment by addb** [comment]
> > I am considering an alternative solution, to modify getSharedState to include empty values when aid/vid are not available, so the shared state and response identity event shares the keys verbosely like:
> > ```
> > Standard shared state created for com.adobe.module.analytics with version x and data: 
> > {
> >   "aid" : "",
> >   "vid" : ""
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This option is also in line with Android and ACP versions. @kevinlind @addb thoughts?
> 
> Honestly, I prefer to filter out null/empty values from the EventData. I think the Swift implementation actually corrects the C++ implementation, so I'd like to leave the shared state as it is. If anything, I would suggest, in IdentityProperties, to set aid/vid to nil if the setter's are passed an empty string. It appears the current implementation could return either nil or empty string to the public API.

I am also of the opinion to not share empty values and instead have nil.

File: AEPAnalytics/Sources/Analytics+PublicAPI.swift [hunk 1]
```diff
@@ -65,10 +65,7 @@ import Foundation
                 return
             }
 
-            guard let trackingIdentifier = responseEvent.data?[AnalyticsConstants.EventDataKeys.ANALYTICS_ID] as? String else {
-                completion(nil, AEPError.unexpected)
-                return
-            }
+            let trackingIdentifier = responseEvent.data?[AnalyticsConstants.EventDataKeys.ANALYTICS_ID] as? String
```
Comment (emdobrin) [inline AEPAnalytics/Sources/Analytics+PublicAPI.swift:68]: We should handle the missing key and key present but value not a String cases. If value is not a String can still be AEPError.unexpected.
Same for visitor identifier

File: AEPAnalytics/Tests/UnitTests/AnalyticsAPITests.swift [hunk 2]
```diff
@@ -202,7 +202,7 @@ class AnalyticsAPITests: XCTestCase {
         // test
         Analytics.getTrackingIdentifier(){(identifier, error) in
             XCTAssertNil(identifier)
-            XCTAssertEqual(error as? AEPError, .unexpected)
+            XCTAssertNil(error)
```
Comment (emdobrin) [inline AEPAnalytics/Tests/UnitTests/AnalyticsAPITests.swift:205]: We can update testGetTrackingIdentifier_IncorrectResponse to handle payload with non String values for example and create new tests for 
testGetTrackingIdentifier_correctResponseWithNilIdentifier()
Same comment for visitorIdentifier API.

**Comment by emdobrin** [comment]
Sounds good, let's make sure the shared state/response event docs are up to date too. 
Left some small comments on the logic.

File: AEPAnalytics/Sources/Analytics+PublicAPI.swift [hunk 3]
```diff
@@ -65,11 +65,18 @@ import Foundation
                 return
             }
 
-            guard let trackingIdentifier = responseEvent.data?[AnalyticsConstants.EventDataKeys.ANALYTICS_ID] as? String else {
-                completion(nil, AEPError.unexpected)
-                return
+            if let responseData = responseEvent.data {
+                if responseData.keys.contains(AnalyticsConstants.EventDataKeys.ANALYTICS_ID) {
```
Comment (emdobrin) [inline AEPAnalytics/Sources/Analytics+PublicAPI.swift:69]: we could group these two into a single if statement 
```
if let responseData = responseEvent.data, responseData.keys.contains(AnalyticsConstants.EventDataKeys.ANALYTICS_ID) 
```
similar for vid

**Review by emdobrin — COMMENTED** [review]
Looks good with a small suggestion

File: AEPAnalytics/Sources/Analytics+PublicAPI.swift (hunk 3 above)
Comment (addb) [reply AEPAnalytics/Sources/Analytics+PublicAPI.swift:69]: Oh yes 👍

**Review by kevinlind — APPROVED** [review]

**Review by emdobrin — APPROVED** [review]

**Pull request merged**